dev/test/document.ts

```diff
@@ -2020,3 +2022,28 @@ describe('listCollections() method', () => {
     });
   });
 });
+
+describe('withConverter() support', () => {
+  let firestore: Firestore;
+
+  beforeEach(() => {
+    return createInstance().then(firestoreInstance => {
+      firestore = firestoreInstance;
+    });
+  });
+
+  afterEach(() => verifyInstance(firestore));
+
+  it('for DocumentReference.withConverter()', async () => {
```

Comment:
Note that your tests do not pass since you need to initialize the mock layer for unit test. This would look something like this:


```
describe('withConverter() support', () => {
  it('for DocumentReference.withConverter()', async () => {
    const doc = document('documentId', 'author', 'author', 'title', 'post');
    const overrides: ApiOverride = {
      commit: (request, options, callback) => {
        const expectedRequest = set({
          document: doc,
        });
        requestEquals(request, expectedRequest);

        callback(null, writeResult(1));
      },
      batchGetDocuments:  () => {
        const stream = through2.obj();
        setImmediate(() => {
          stream.push({found: doc, readTime: {seconds: 5, nanos: 6}});
          stream.push(null);
        });

        return stream;
      }
    };

    return createInstance(overrides).then(async firestore => {
      const docRef = firestore
          .collection('collectionId')
          .doc('documentId')
          .withConverter(postConverter);

      await docRef.set(new Post('post', 'author'));
      const postData = await docRef.get();
      const post = postData.data();
      expect(post).to.not.equal(undefined);
      expect(post!.byline()).to.equal('post, by author');
    });
  });
});
```

Reply:
Done.